frontend/dialob-composer-material/src/components/PageTabs.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { Menu, MenuItem, Button, IconButton, Box, TextField, Table, TableRow, TableBody, TableContainer, CircularProgress, Typography } from '@mui/material';
+import { Add, Close, ContentCopy, Key, Menu as MenuIcon, Tune, Visibility } from '@mui/icons-material';
+import { DialobItem, DialobItems, useEditor } from "../dialob";
+import { FormattedMessage, useIntl } from "react-intl";
+
+
+const MAX_PAGE_NAME_LENGTH = 40;
+
+const getPageTabTitle = (item: DialobItem, language: string): string => {
+  const rawLabel = item.label ? item.label[language] : null;
+  if (!rawLabel) {
+    return item.id;
+  } else {
+    return rawLabel.length > MAX_PAGE_NAME_LENGTH
+      ? rawLabel.substring(0, MAX_PAGE_NAME_LENGTH) + '…'
+      : rawLabel;
+  }
+};
+
+const PageHeader: React.FC<{ item?: DialobItem }> = ({ item }) => {
+  const intl = useIntl();
+  if (!item) {
+    return null;
+  }
+  return (
+    <TableContainer sx={{ borderRadius: 2, borderTopLeftRadius: 0, border: 1, borderColor: 'divider' }}>
+      <Table>
+        <TableBody>
+          <TableRow sx={{ borderBottom: 1, borderColor: 'divider' }}>
+            <TextField
```

Comment:
All labels in forms must not be editable "inline" like in old composer,  but through a separate editor window as described in the "spec" presentation. This is one of the main reasons for the rewrite.